### PR TITLE
Update SSD documentation.

### PR DESCRIPTION
**Pull request by xingyuanbu**
See https://github.com/sefira/models/tree/ssd_wiki/fluid/object_detection

### CODE REVIEW

File: fluid/object_detection/README.md [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+[Single Shot MultiBox Detector (SSD)](https://arxiv.org/abs/1512.02325) framework for object detection can be categorized as a single stage detector. A single stage detector simplifies object detection as a regression problem, which directly predicts the bounding boxes and class probabilities without region proposal. SSD further makes improves by producing these predictions of different scales from different layers, as shown below. SSD is readily pluggable into to a wide variant standard convolutional network, such as VGG, ResNet, or MobileNet, which is also called base network or backbone. In this tutorial we used [MobileNet](https://arxiv.org/abs/1704.04861).
+<p align="center">
+<img src="images/SSD_paper_figure.jpg" height=300 width=900 hspace='10'/> <br />
+The Single Shot MultiBox Detector (SSD)
+</p>
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: 1. 需要简要的解释下图中的SSD结构。
2. In this tutorial we used [MobileNet](https://arxiv.org/abs/1704.04861).  这句话放在图的后面，引出下文。

File: fluid/object_detection/README.md [hunk 2]
```diff
@@ -50,31 +54,24 @@ Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https
     ```
 
 #### Train on PASCAL VOC
-  - Train on one device (/GPU).
-  ```python
-  env CUDA_VISIBLE_DEVICES=0 python -u train.py --parallel=False --dataset='pascalvoc' --pretrained_model='pretrained/ssd_mobilenet_v1_coco/'
-  ```
-  - Train on multi devices (/GPUs).
-
+train.py is the main caller of the training module. Examples of usage are shown below.
```
Comment (qingqing01) [inline fluid/object_detection/README.md]:
```
train.py  -> `train.py` 
```

File: fluid/object_detection/README.md [hunk 3]
```diff
@@ -50,31 +54,24 @@ Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https
...
   ```python
-  env CUDA_VISIBLE_DEVICES=0,1 python -u train.py --batch_size=64 --dataset='pascalvoc' --pretrained_model='pretrained/ssd_mobilenet_v1_coco/'
+  python -u train.py --batch_size=64 --dataset='pascalvoc' --pretrained_model='pretrained/ssd_mobilenet_v1_coco/'
   ```
+   - Set ```env CUDA_VISIBLE_DEVICES=0,1``` to specifiy the number of GPU you want to use.
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: env CUDA_VISIBLE_DEVICES=0,1 
->
export CUDA_VISIBLE_DEVICES=0,1

File: fluid/object_detection/README.md [hunk 4]
```diff
@@ -50,31 +54,24 @@ Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https
...
+   - Set ```--dataset='coco2014'``` or ```--dataset='coco2017'``` to train model on MS COCO dataset.
+   - For more configurations, see the arguments list in train.py.
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: > For more configurations, see the arguments list in train.py.

改成下面吧：

```
- For more help on arguments:
    ```bash
    python train.py --help
    ```
```

File: fluid/object_detection/README.md [hunk 5]
```diff
@@ -50,31 +54,24 @@ Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https
...
-  - Train on one device (/GPU).
-  ```python
-  env CUDA_VISIBLE_DEVICES=0 python -u train.py --parallel=False --dataset='pascalvoc' --pretrained_model='pretrained/ssd_mobilenet_v1_coco/'
-  ```
-  - Train on multi devices (/GPUs).
-
+train.py is the main caller of the training module. Examples of usage are shown below.
   ```python
-  env CUDA_VISIBLE_DEVICES=0,1 python -u train.py --batch_size=64 --dataset='pascalvoc' --pretrained_model='pretrained/ssd_mobilenet_v1_coco/'
+  python -u train.py --batch_size=64 --dataset='pascalvoc' --pretrained_model='pretrained/ssd_mobilenet_v1_coco/'
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: ```python  -> ```bash

文档中所有命令行运行的地方，都改下吧。

File: fluid/object_detection/README.md [hunk 6]
```diff
@@ -50,31 +54,24 @@ Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https
...
-#### Train on MS-COCO
-  - Train on one device (/GPU).
-  ```python
-  env CUDA_VISIBLE_DEVICES=0 python -u train.py --parallel=False --dataset='coco2014' --pretrained_model='pretrained/mobilenet_v1_imagenet/'
-  ```
-  - Train on multi devices (/GPUs).
-  ```python
-  env CUDA_VISIBLE_DEVICES=0,1 python -u train.py --batch_size=64 --dataset='coco2014' --pretrained_model='pretrained/mobilenet_v1_imagenet/'
-  ```
+### Evaluate
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: Evaluate前面，Train的最后需要说明下，使用的优化方法、学习率调整策略、正则等，以及训练了多少epoc，得到的结果是多少等。

File: fluid/object_detection/README.md [hunk 7]
```diff
@@ -86,36 +83,21 @@ make install
...
-```python
-env CUDA_VISIBLE_DEVICES=0 python eval.py --dataset='coco2014' --nms_threshold=0.5 --model_dir='train_coco_model/40' --test_list='annotations/instances_minival2014.json' --ap_version='integral'
-env CUDA_VISIBLE_DEVICES=0 python eval_coco_map.py --dataset='coco2017' --nms_threshold=0.5 --model_dir='train_coco_model/40' --test_list='annotations/instances_minival2017.json'
-```
-
-TBD
 
 ### Infer and Visualize
-
+infer.py is the main caller of the infering module. Examples of usage are shown below.
```
Comment (qingqing01) [inline fluid/object_detection/README.md]:
```
`infer.py` is the main ...
```

File: fluid/object_detection/README.md [hunk 8]
```diff
@@ -86,36 +83,21 @@ make install
...
 Below is the examples after running python infer.py to inference and visualize the model result.
 <p align="center">
-<img src="images/COCO_val2014_000000000139.jpg" height=300 width=400 hspace='10'/>
-<img src="images/COCO_val2014_000000000785.jpg" height=300 width=400 hspace='10'/>
-<img src="images/COCO_val2014_000000142324.jpg" height=300 width=400 hspace='10'/>
-<img src="images/COCO_val2014_000000144003.jpg" height=300 width=400 hspace='10'/> <br />
+<img src="images/009943.jpg" height=300 width=400 hspace='10'/>
+<img src="images/009956.jpg" height=300 width=400 hspace='10'/>
+<img src="images/009960.jpg" height=300 width=400 hspace='10'/>
+<img src="images/009962.jpg" height=300 width=400 hspace='10'/> <br />
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: 图像的类别name可以再调大点。

File: fluid/object_detection/README.md [hunk 9]
```diff
@@ -36,9 +40,9 @@ cd data/coco
...
+We provide two pre-trained models. The one is MobileNet-v1 SSD trained on COCO dataset, but removed the convolutional predictors for COCO dataset. This model can be used to initialize the models when training other datasets, like PASCAL VOC. The other pre-trained model is MobileNet v1 trained on ImageNet 2012 dataset but removed the last weights and bias in the Fully-Connected layer.
 
-Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https://github.com/tensorflow/models/blob/f87a58cd96d45de73c9a8330a06b2ab56749a7fa/research/object_detection/g3doc/detection_model_zoo.md). The MobileNet v1 model is converted [Caffe](https://github.com/shicai/MobileNet-Caffe).
+Declaration: the MobileNet-v1 SSD model is converted by [TensorFlow model](https://github.com/tensorflow/models/blob/f87a58cd96d45de73c9a8330a06b2ab56749a7fa/research/object_detection/g3doc/detection_model_zoo.md). The MobileNet v1 model is converted from [Caffe](https://github.com/shicai/MobileNet-Caffe).
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: 这里的声明要么加上：我们将来会换成自己训练的模型？

File: fluid/object_detection/README.md [hunk 10]
```diff
@@ -86,36 +83,21 @@ make install
...
-TBD
 
 ### Infer and Visualize
-
+infer.py is the main caller of the infering module. Examples of usage are shown below.
 ```python
-env CUDA_VISIBLE_DEVICES=0 python infer.py --dataset='coco' --nms_threshold=0.5 --model_dir='train_coco_model/20' --image_path='./data/coco/val2014/COCO_val2014_000000000139.jpg'
+python infer.py --dataset='pascalvoc' --nms_threshold=0.45 --model_dir='train_pascal_model/best_model' --image_path='./data/pascalvoc/VOCdevkit/VOC2007/JPEGImages/009963.jpg'
 ```
 Below is the examples after running python infer.py to inference and visualize the model result.
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: Below is the examples after running python infer.py to inference and visualize the model result.

->

Below are the examples of running the inference and visualizing the model result.

File: fluid/object_detection/README.md [hunk 11]
```diff
@@ -125,5 +107,3 @@ TBD
 |MobileNet-v1-SSD 300x300  | COCO MobileNet SSD | VOC07+12 trainval| VOC07 test   | xx%  |
```
Comment (qingqing01) [inline fluid/object_detection/README.md]: 去掉表格里的最后两行吧，填下第一行的mAP结果。

**Review by qingqing01 — COMMENTED** [review]
@sefira Thanks very much!

File: fluid/object_detection/README.md (hunk 1 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 2 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 4 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 5 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 7 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 8 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 9 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 10 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

File: fluid/object_detection/README.md (hunk 3 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md]: revised

**Review by qingqing01 — APPROVED** [review]

File: fluid/object_detection/README.md [hunk 12]
```diff
@@ -1,20 +1,27 @@
...
+</p>
+
+SSD is readily pluggable into a wide variant standard convolutional network, such as VGG, ResNet, or MobileNet, which is also called base network or backbone. In this tutorial we used [MobileNet](https://arxiv.org/abs/1704.04861).
+
 
 ### Data Preparation
 
 You can use [PASCAL VOC dataset](http://host.robots.ox.ac.uk/pascal/VOC/) or [MS-COCO dataset](http://cocodataset.org/#download).
 
 #### PASCAL VOC Dataset
```
Comment (qingqing01) [inline fluid/object_detection/README.md:22]: Remove `#### PASCAL VOC Dataset`.

File: fluid/object_detection/README.md [hunk 13]
```diff
@@ -25,7 +32,7 @@ The command `download.sh` also will create training and testing file lists.
 
 #### MS-COCO Dataset
```
Comment (qingqing01) [inline fluid/object_detection/README.md:33]: Remove ` #### MS-COCO Dataset`.

File: fluid/object_detection/README.md [hunk 14]
```diff
@@ -1,20 +1,27 @@
-The minimum PaddlePaddle version needed for the code sample in this directory is the lastest develop branch. If you are on a version of PaddlePaddle earlier than this, [please update your installation](http://www.paddlepaddle.org/docs/develop/documentation/en/build_and_install/pip_install_en.html).
+The minimum PaddlePaddle version needed for the code sample in this directory is the latest develop branch. If you are on a version of PaddlePaddle earlier than this, [please update your installation](http://www.paddlepaddle.org/docs/develop/documentation/en/build_and_install/pip_install_en.html).
 
 ---
 
 ## SSD Object Detection
```
Comment (qingqing01) [inline fluid/object_detection/README.md:5]: Add `Table of Contents` before line 5, refer https://github.com/PaddlePaddle/models/pull/979/files#diff-e74a88ec056840ea1a6d958b791284afR5

File: fluid/object_detection/README.md [hunk 15]
```diff
@@ -36,45 +43,46 @@ cd data/coco
...
-  ```
-  - Train on multi devices (/GPUs).
-  ```python
-  env CUDA_VISIBLE_DEVICES=0,1 python -u train.py --batch_size=64 --dataset='coco2014' --pretrained_model='pretrained/mobilenet_v1_imagenet/'
+  ```bash
+  python train.py --help
   ```
 
-TBD
+We used RMSProp optimizer with mini-batch size 64 to train the MobileNet-SSD. The initial learning rate is 0.001, and was decayed at 40, 60, 80, 100 epochs with multiplier 0.5, 0.25, 0.1, 0.01, respectively. Weight decay is 0.00005. After 120 epochs we achive XXX% mAP under 11point metric.
```
Comment (qingqing01) [inline fluid/object_detection/README.md:74]: Introduction the data argumentation and data reader before line 74.  refer https://github.com/PaddlePaddle/models/pull/979/files#diff-e74a88ec056840ea1a6d958b791284afR87

File: fluid/object_detection/README.md (hunk 12 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md:22]: revised

File: fluid/object_detection/README.md (hunk 13 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md:33]: revised

File: fluid/object_detection/README.md (hunk 14 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md:5]: revised

File: fluid/object_detection/README.md (hunk 15 above)
Comment (xingyuanbu) [reply fluid/object_detection/README.md:74]: revised

**Pull request merged**